src/data/glossary.yaml

```diff
@@ -89,12 +107,31 @@
     title: Measure Performance with the RAIL Model
     link: /web/fundamentals/performance/rail
 
+- term: Responsive Design
+  description: "Responsive web design is an approach to web design aimed at making web pages visually appealing and performant on any form factor. In addition, it is important to understand that responsive web design tasks include offering the same content to a variety of devices for a single website."
+  see:
+    title: Responsive Web Design Basics
+    link: /web/fundamentals/design-and-ux/responsive/
+
+- term: Service Worker
+  description: "Service workers essentially act as proxy servers that sit between web applications, and the browser and network (when available). They are intended to (amongst other things) enable the creation of effective offline experiences, intercepting network requests and taking appropriate action based on whether the network is available and updated assets reside on the server. They also allow access to push notifications and background sync APIs."
+  links:
+  - title: Service Worker API
+    link: https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API
+
 - term: Shadow DOM
   description: "Shadow DOM introduces scoped CSS and DOM to the web platform. It lets developers write encapsulated UI components which can be used in any application."
   link: /web/fundamentals/architecture/building-components/shadowdom
 
 - term: Single Page App
   acronym: SPA
+  description: "Snappy user-friendly apps that perform more like desktop apps. Typically, SPA content is rendered dynamically using JavaScript, rather than opening a new page. As a result, single-page applications typically load only data (e.g. in JSON or XML format) rather than pre-rendered HTML. This decreases the data transferred on the wire. Google's headline web apps—such as GMail, Inbox, Maps, Docs, Sheets, and so on—have been pioneers here."
```

Comment:
I'd remove "snappy" and anything else that is overly promotional. While it's often true, it's not always true.